cargo-fix/src/cli.rs

```diff
@@ -77,6 +84,15 @@ pub fn run() -> Result<(), Error> {
         cmd.env("RUSTC_ORIGINAL", rustc);
     }
 
+    // Trigger edition-upgrade mode. Currently only supports the 2018 edition.
+    info!("edition upgrade? {:?}", matches.value_of("edition"));
+    if let Some("2018") = matches.value_of("edition") {
+        info!("edition upgrade!");
+        let mut rustc_flags = env::var_os("RUSTFLAGS").unwrap_or_else(|| "".into());
+        rustc_flags.push("-W rust-2018-breakage -A unused -A nonstandard-style -A bad-style");
```

Comment:
How come these specific `-A` flags were required? I'd actually sort of expect that we don't pass any extra arguments other than the `-W` for the lint

Reply:
Hm is it bad though if other warnings are emitted? Or is it bad if we fix some otherwise fixable warnings along the way? 

I'm worried about this being an ever-growing list :(

Reply:
I'm all for fixing as many things as possible, but it might send the wrong message in this specific context. E.g., if I run `cargo fix --prepare-for 2018` and it removes a bunch of (unused) `mut`s I might end up thinking that this is required for the new edition.

I'll try and play around with filtering by the lint group that triggered a lint. If that's possible we can just add the `-W rust-2018-breakage` and skip fixing other lints.

Reply:
True yeah we could filter, but I'd imagine that for most maintained codebases they're already warning-free, so I wouldn't expect something like this to come up much in practice

Reply:
I didn't even mean filtering, just abort if there are any (and tell the user the reason for the abort)

Reply:
Oh sure yeah that's a possibility, although I'm still not sure why we would go out of our way to worry about other lints. Everything should work A-OK if we *purely* pass `-W rust-2018-breakage`. The user will be informed, as they always are, of lints that couldn't be fixed. Anything that comes up which is fixable as part of the breakage lint is already fixed.